jina/drivers/control.py

```diff
@@ -104,6 +103,22 @@ def __call__(self, *args, **kwargs):
         else:
             raise UnknownControlCommand(f'don\'t know how to handle {self.req.command}')
 
+    def _dump(self):
+        if self.req.targets and self.runtime.__class__.__name__ == 'ZEDRuntime':
+            patterns = self.req.targets
+            if isinstance(patterns, str):
+                patterns = [patterns]
+            for p in patterns:
+                if re.match(p, self.runtime.name):
+                    self.logger.info(
+                        f'Dumping from Executor `{self.runtime._executor.name}` in `{self.runtime.name}`'
+                    )
+                    req_dict = dict(self.req.args)
+                    self.runtime._executor.dump(
```

Comment:
code smell accessing the private attribute of runtime.

If dump needs access to Executor, maybe we need to rethink if it should be a ControlRequest

Reply:
or have ControlReqDriver implement BaseExecutableDriver interface

Reply:
Initially it was a different Request, but it does make sense that it's just another ControlRequest, as it doesn't behave in an inherently different way. It just has a bunch of args (which ControlReq always had but were never used). It would make sense that the Driver is attached to an executor directly. Not sure what the vision is for this change Han/Max mentioned @maximilianwerk ?

Reply:
what is clear is that this Driver needs to implement BaseExecutableDriver. this changes challenges the design of ControlDriver right now.

Reply:
To be honest, I don't yet have a clear view of the `ControlDriver` after the API refactoring. It might just not be much affected. It might also be joined with the `GenericDriver`. In any case, I am OK with this hack until we understand the design of the new Executor handling in detail.

Reply:
so lets make a huge TODO and clarify that is a hack to be checked later

Reply:
added a TODO there